### DISCUSSION TITLE
Troubleshooting SA1 clock

### DISCUSSION THREAD

**Discussion by kdawg8** [sanni/cartreader#173] [Need help]
After many attempts to read Super Mario RPG with no luck, I loaded the clock calibration sketch and after connecting the two pins together in the SNES socket I'm getting 0 MHz.  I've verified all of the solder joints are good and the cart reader works fine with all other cartridge types for SNES and other consoles.  How can I troubleshoot the clock generator vs. the snesCIC as causing the issue?  (I have a BK 390a multimeter which can measure frequency for troubleshooting)  Thanks!

**Comment by sanni** [comment]
Release V6.1 had a bug where the clockgen could not initialize, please try V6.4.
I also updated the [clkcalc sketch](https://www.dropbox.com/s/uyhpnng9zgfl2ja/clkcal.zip?dl=1) to support exFAT.

Clkcal sketch:

Set the config switches like this:

![](https://dl.dropboxusercontent.com/s/rj9rf0dcyzeespw/snes_jumper.jpg?dl=1)   

Then press one end of a jumper wire against SNES slot pin 43(A18/BA2) and the other end of the wire against one of the following pins and it should read those values: 
- 3072000.00 Hz when connected to SNES slot pin 56(CIC Clock) 
- 4000000.00 Hz when connected to SNES slot pin 1(EXT Clock) 
- 1000000.00 Hz when connected to SNES slot pin 57(CPU Clock) 

![](https://i.imgur.com/srywncH.jpg) 

![](https://i.imgur.com/Dpu4ivo.jpg) 

Also make sure the clock generator is installed with the parts side up.

![](https://i.imgur.com/LMdS7jX.jpg) 

If it still shows 0 Mhz in the clkcal sketch then the clock generator could be broken.

**Reply by kdawg8** [reply]
Connecting pin 43 to each of the 3 pins you indicated gives, after a brief transient, 0 MHz.  I also connected pin 43 to the Arduino SCL pin to check that there weren't any issues with uploading the clkcal sketch and I get 77,800 Hz (is this right?  seems low).  While doing this I realized I haven't checked the CLK0 or CLK1 switches yet for good solder joints, will do that once I get a new battery for my meter.  If those are good I guess I'll order a new clock generator.  Thanks for your help!

**Comment by kdawg8** [comment]
Got a new clock generator in today, clkcal is showing the correct frequencies now and I was able to dump my childhood saves from Super Mario RPG : )

**Comment by sanni** [comment]
Thanks for reporting back. Happy it now works for you.

**Comment by Chomemel** [comment]
Hi,
I write here because I don't really understand how to calibrate the clock.

Frequency is roughly 3093600 Hz between pin 43 and 56.
![IMG_20220713_182115](https://user-images.githubusercontent.com/107198559/178787450-6f02fbed-714f-4c0e-904d-d59a943e4aca.jpg)

If I press one switch, I see **"adjusting"**. Then **"correction"** is set the value of adjustement. But the **"frequency"** is still the same. See.
![IMG_20220713_182237](https://user-images.githubusercontent.com/107198559/178787908-6ebb6c95-4d76-4afb-b5a2-56b3ecac81b8.jpg)

I also checked with pin 1 and pin 57. And here is what I read.
![IMG_20220713_182333](https://user-images.githubusercontent.com/107198559/178788118-d58f7af7-b230-4604-9fb7-4a9e35399481.jpg)
![IMG_20220713_182359](https://user-images.githubusercontent.com/107198559/178788339-c60fd718-1b29-4e91-8bfd-7335921c7981.jpg)

**Comment by Chomemel** [comment]
Another question.
Are the clock generator and the cic only for SA-1 games ? Or also for the ones with other enhancement chips (like FX or CX4) ?

**Comment by sanni** [comment]
SI5351 Clock Gen is for SNES SPC7110/SA1/SF Memory/Satellaview cartridges and N64 cartridges with EEPROM saves.
The snesCIC is only used for SA1 afaik.

Try calibrating the Clock Generator by removing "//" in front of "#define clockgen_calibration" and "#define clockgen_installed" in Cart_Reader.ino and then selecting "Calibrate Clock" in the SNES submenu.
There you can click a button to increase the step width, maybe you just need a higher offset.

**Comment by Chomemel** [comment]
Thanks Sanni for the answers.

I updated with 9.1 and changed what you told me in the options. I was able to calibrate the clock form the SNES options. I guess, I needed a higher offset indeed.

Unfortunately, SA1 games still can't be read.

I tried reading N64 cartridge with EEPROM and both ROM and saves can be read. So I guess the clock generator works fine (at least "CLK1"). 

So I have 2 things left.
- The super cic
- CLK0 of the clock gen (but if CLK1 works, CLK0 should wor too, right?)

**Reply by Chomemel** [reply]
Ok, I might have found the problem. I'll update when I'll be able to verify

**Comment by Chomemel** [comment]
Finaly, I fixed everything. First of all I was stupid enough to solder the PIC upside down. Then I also realised that I flashed the wrong hex file. What a dumb.
Anyway, I was able to dump ROM and files from SA-1 cartrige. However, the cart reader can be a bit more picky with SA-1 cartridge...

Here are pictures of triumph
![IMG_20220823_002052](https://user-images.githubusercontent.com/107198559/186535982-6dff3807-2a69-4b03-ac81-5fa176a81227.jpg)
![IMG_20220823_002113](https://user-images.githubusercontent.com/107198559/186536553-fdf5f360-aecc-4840-a0c5-30bd35712678.jpg)

Thanks a lot again Sanni. Now the elec part is over.

**Comment by superbonaci** [comment]
I'm wondering if the hardware from this link is related to this project: [Open Source Cartridge Reader V3-ALTER Build service [Group F]](https://savethehero.builders/collections/build-service/products/open-source-cartridge-reader-v3-alter-build-service)

since it has a purchase option "SA-1 UNLOCK SNES & N64 EEPROM SAVE (OPTION FEE $28 [PIC12F629 MICROCONTROLLER WITH THE SNESCIC AND CLOCK GENERATOR])"

If I have such board borrowed from a friend, how can I tell if it was purchased with the SA-1 option or not? Any way to tell without any physical game?

**Reply by Chomemel** [reply]
There should be the 12F629 microcontroller soldered and the clock genreator.
You should check [this link](https://github.com/sanni/cartreader/wiki/How-to-build-the-PCB) to see what they look like. Check out the video too.

**Reply by superbonaci** [reply]
@Chomemel thanks!
